components/toast/script.js

```diff
@@ -0,0 +1,174 @@
+
+    get position() {
+        return this._position;
+    }
+
+    set position(value) {
+        this._position = value;
+        this.setAttribute('position', value);
+    }
+
+    get gravity() {
+        return this._gravity;
+    }
+
+    set gravity(value) {
+        this._gravity = value;
+        this.setAttribute('gravity', value);
+    }
+
+    init(data) {
+        this.setupTemplate(data, () => {
+            components.renderOnce(this);
+            // attach event handlers here
+            this.attachEventListeners();
+            this._messageSlot = this.querySelector('.guic-toast-message').firstElementChild;
+        });
+    }
+
+    connectedCallback() {
+        this.gravity = this.getAttribute('gravity') || this._gravity;
+        this.position = this.getAttribute('position') || this._position;
+        this.timeout = this.getAttribute('timeout') || 0;
+        this.elementSelector = this.getAttribute('target');
+        this._targetElement = this._targetElement || document.querySelector(this.elementSelector);
+
+        if (!containersCreated) this.createToastContainers();
+
+        components.loadResource(this)
+            .then(this.init)
+            .catch(err => console.error(err));
+    }
+
+    attachEventListeners() {
+        if (this._targetElement ) {
+            this._targetElement.addEventListener('click', this.show);
+        }
+
+        if (!animationEventAttached) {
+            document.addEventListener('animationend', (event) => {
+                if (event.animationName === 'guic-toast-fade-out') event.target.parentElement.removeChild(event.target);
+            });
+            animationEventAttached = true;
+        }
+    }
+    /* eslint-enable require-jsdoc */
+
+    /**
+     * Creates containers for all possible toast positions
+    */
+    createToastContainers() {
+        const body = document.querySelector('body');
+        TOAST_POSITIONS.forEach((containerPosition) => {
+            const [vertical, horizontal] = containerPosition.split(' ');
+            const toastContainer = document.createElement('div');
+            toastContainer.classList.add('guic-toast-container', `${classPrefix}${vertical}`, `${classPrefix}${horizontal}`);
+            body.appendChild(toastContainer);
+        });
+
+        containersCreated = true;
+    }
+
+    /**
+     * Appends the toast to one of the containers depending on the gravity and position
+     * @param {string} gravity - top, bottom.
+     * @param {string} position - left, right, center
+    */
+    appendToastToContainer(gravity, position) {
+        const container = document.querySelector(`.guic-toast-container.${classPrefix}${gravity}.${classPrefix}${position}`);
+
+        if (container) {
+            container.appendChild(this);
+            return;
+        } else {
+            console.error('No container found for the specified gravity and position');
+        }
+    }
+
+    /**
+     * Displays the toast
+     */
+    show() {
```

Comment:
Make sure when the element is shown that you are not able to trigger the show function again.

Reply:
Fixed